### ISSUE TITLE
bedMethyl index missing in dmr multi with replicates 

### ISSUE THREAD

**Issue by Yang990-sys** [nanoporetech/modkit#294]
After following your guidance in #143 , we have made the following attempts:
![image](https://github.com/user-attachments/assets/433e9737-0109-4eb1-90d1-dabaa76a1d80)

while, all regions failed for the following reason:
![image](https://github.com/user-attachments/assets/61b0d95d-e80e-490d-a1cf-0eb8601f1761)

In order to check for file errors, I gave different sample names to four samples, and the results were normal：
![image](https://github.com/user-attachments/assets/1f6ca543-ff18-400d-962a-472c1593400d)

Is there any way to solve my problem？

**Comment by Yang990-sys** [comment]
my modkit version : 
![image](https://github.com/user-attachments/assets/72e996d3-d28d-4acc-85bf-c1151598d82d)

**Comment by Yang990-sys** [comment]
After regenerating the index file, the issue was resolved

**Comment by ArtRand** [comment]
Hello @Yang990-sys, 

Thanks for finding this. I think `modkit dmr` should make the folder if it doesn't exist to align it's behavior with the rest of the commands. I'll make sure to add this to the next release.

**Comment by ArtRand** [comment]
I'm going to keep this open to track the work. Thanks again for noticing the inconsistency.

**Comment by Yang990-sys** [comment]
The reason I found seems inaccurate because I encountered this error again. Looking forward to software updates

**Comment by ArtRand** [comment]
Hello @Yang990-sys,

Could you attach the log file for the command that is failing to this thread? Also could you run

```bash
tabix -l ${sample}
```

for each of the samples you have and attach that as well?

**Comment by Yang990-sys** [comment]
Because there is no problem running for a single sample, it should not be an issue with the index itself. When there is a duplicate, the first step of the program is to merge the duplicate samples. So I guess it may be because the merged file cannot use the original index?

**Comment by Yang990-sys** [comment]
![image](https://github.com/user-attachments/assets/33cd2992-133f-4699-98ec-6241b322f844)
[out.log](https://github.com/user-attachments/files/18003760/out.log)

**Comment by ArtRand** [comment]
@Yang990-sys Just to be sure. All of the pileups use the same reference, which in this case is `all_transcripts.fa`, correct?

**Comment by Yang990-sys** [comment]
Yes, we have encountered the same problem in multiple batches of data

**Comment by Yang990-sys** [comment]
I wrote a python script for merging bedmethyl files, and I want to know if it's scientific to merge first and then run `modkit dmr`

**Comment by ArtRand** [comment]
@Yang990-sys would you be willing to share the data that exposes the problem with me? We can do it through email so that it's not publicly accessible. My email is art.rand [at] nanoporetech.com
